pike/test/__init__.py

```diff
@@ -156,7 +186,36 @@ def default_client(signing=None):
 @attr.s
 class TreeConnect(object):
     """
-    Combines a Client, Connection, Channel, and Tree for simple access to an SMB share.
+    Combines a :py:class:`~pike.model.Client`,
+    :py:class:`~pike.model.Connection`, :py:class:`~pike.model.Channel`, and
+    :py:class:`~pike.model.Tree` for simple access to an SMB share.
+
+    May be used as a contextmanager: entering the context will establish the
+    connection, session, and tree, and exiting the context will cleanly
+    disconnect (if possible). Errors on disconnect are ignored.
+
+    Any values not specified in the constructor will be drawn from the
+    associated environment variables. At minimum the following should be
+    specified explicitly or in the environment:
+
+      * ``server`` (``PIKE_SERVER``) - server or hostname to connect to
+      * ``creds`` (``PIKE_CREDS``) - percent-delimited credential string
+      * ``share`` (``PIKE_SHARE``) - share name to connect to
+
+    All other arguments are optional.
+
+    The ``TreeConnect`` object can be used with the division operator, ``/``,
+    which returns a :py:class:`~pike.path.PikePath` instance representing the
+    path extended from the share root. The tree connection will be established
+    before returning the :py:class:`~pike.path.PikePath` if is is not already
+    connected.
+
```

Comment:
A code snippet in the docstring would be great

    # print entries under a share path
    with TreeConnect(...) as tc:
        for entry in tc:
            if entry.is_file():
                print("file :{}".format(entry.path)
            elif entry.is_dir():
                print("dir  :{}".format(entry.path)
            else:
                print("other:{}".format(entry.path)

(I made up that code; you get the idea)